src/mono/wasm/runtime/loader/polyfills.ts

```diff
@@ -1,11 +1,11 @@
 
 import type { DotnetModuleInternal } from "../types/internal";
-import { INTERNAL, ENVIRONMENT_IS_NODE, ENVIRONMENT_IS_SHELL, loaderHelpers } from "./globals";
+import { INTERNAL, ENVIRONMENT_IS_NODE, ENVIRONMENT_IS_SHELL, loaderHelpers, ENVIRONMENT_IS_WEB } from "./globals";
 
 let node_fs: any | undefined = undefined;
 let node_url: any | undefined = undefined;
 
-export async function init_polyfills(module: DotnetModuleInternal): Promise<void> {
+export async function detect_features_and_polyfill(module: DotnetModuleInternal): Promise<void> {
```

Comment:
@kg we could start adding more feature detection here on next PRs, like SIMD, MT etc